### ISSUE TITLE
[CLI] help command not working for optimizer and lr_scheduler

### ISSUE THREAD

**Issue by F-Barto** [PyTorchLightning/pytorch-lightning#10725]
## 🐛 Bug

Hello everyone and many thanks for your awesome work. It seems that the commands `--optimizer.help CLASS` and `--lr_scheduler.help CLASS` do not work (at least not as described).

### To Reproduce

```python
#test_help.py

import pytorch_lightning as pl
from pytorch_lightning.utilities.cli import LightningCLI

from typing import List

class MyModel(pl.LightningModule):
    def __init__(self, encoder_layers: int = 12, decoder_layers: List[int] = [2, 4]):
        """Example encoder-decoder model

        Args:
            encoder_layers: Number of layers for the encoder
            decoder_layers: Number of layers for each decoder block
        """
        super().__init__()


cli = LightningCLI(MyModel)
```

### Expected behavior

The command `python test_help.py fit --help` indicates:
```
Implements Averaged Stochastic Gradient Descent:
  --optimizer.help CLASS
                        Show the help for the given subclass of {ASGD, Adadelta, Adagrad, Adam, AdamW, Adamax, LBFGS, NAdam, RAdam, RMSprop, Rprop, SGD, SparseAdam} and exit.
  --optimizer {"class_path":...[,"init_args":...]}
                        Dictionary with "class_path" and "init_args" for any subclass of (<class 'torch.optim.asgd.ASGD'>, <class 'torch.optim.adadelta.Adadelta'>, <class 'torch.optim.adagrad.Adagrad'>, <class
                        'torch.optim.adam.Adam'>, <class 'torch.optim.adamw.AdamW'>, <class 'torch.optim.adamax.Adamax'>, <class 'torch.optim.lbfgs.LBFGS'>, <class 'torch.optim.nadam.NAdam'>, <class
                        'torch.optim.radam.RAdam'>, <class 'torch.optim.rmsprop.RMSprop'>, <class 'torch.optim.rprop.Rprop'>, <class 'torch.optim.sgd.SGD'>, <class 'torch.optim.sparse_adam.SparseAdam'>).
                        (type: Union[ASGD, Adadelta, Adagrad, Adam, AdamW, Adamax, LBFGS, NAdam, RAdam, RMSprop, Rprop, SGD, SparseAdam])

Chains list of learning rate schedulers. It takes a list of chainable learning:
  --lr_scheduler.help CLASS
                        Show the help for the given subclass of {ChainedScheduler, ConstantLR, CosineAnnealingLR, CosineAnnealingWarmRestarts, CyclicLR, ExponentialLR, LambdaLR, LinearLR, MultiStepLR,
                        MultiplicativeLR, OneCycleLR, SequentialLR, StepLR} and exit.
  --lr_scheduler {"class_path":...[,"init_args":...]}
                        Dictionary with "class_path" and "init_args" for any subclass of (<class 'torch.optim.lr_scheduler.ChainedScheduler'>, <class 'torch.optim.lr_scheduler.ConstantLR'>, <class
                        'torch.optim.lr_scheduler.CosineAnnealingLR'>, <class 'torch.optim.lr_scheduler.CosineAnnealingWarmRestarts'>, <class 'torch.optim.lr_scheduler.CyclicLR'>, <class
                        'torch.optim.lr_scheduler.ExponentialLR'>, <class 'torch.optim.lr_scheduler.LambdaLR'>, <class 'torch.optim.lr_scheduler.LinearLR'>, <class 'torch.optim.lr_scheduler.MultiStepLR'>,
                        <class 'torch.optim.lr_scheduler.MultiplicativeLR'>, <class 'torch.optim.lr_scheduler.OneCycleLR'>, <class 'torch.optim.lr_scheduler.SequentialLR'>, <class
                        'torch.optim.lr_scheduler.StepLR'>). (type: Union[ChainedScheduler, ConstantLR, CosineAnnealingLR, CosineAnnealingWarmRestarts, CyclicLR, ExponentialLR, LambdaLR, LinearLR, MultiStepLR,
                        MultiplicativeLR, OneCycleLR, SequentialLR, StepLR])
```

So, if I understood correctly, with either of these commands, the CLI should give me the corresponding help message (generated from the docstring and signature of the given class):
`python test_help.py fit --optimizer.help Adam`
`python test_help.py fit --optimizer.help torch.optim.adam.Adam`

However, I get the following error:
```
Traceback (most recent call last):
  File "/home/clear/fbartocc/depth_project/LiDARTouch-Refactor/test_help.py", line 21, in <module>
    cli = LightningCLI(MyModel)
  File "/home/clear/fbartocc/miniconda3/envs/Refactor/lib/python3.9/site-packages/pytorch_lightning/utilities/cli.py", line 489, in __init__
    self.parse_arguments(self.parser)
  File "/home/clear/fbartocc/miniconda3/envs/Refactor/lib/python3.9/site-packages/pytorch_lightning/utilities/cli.py", line 628, in parse_arguments
    self.config = parser.parse_args()
  File "/home/clear/fbartocc/miniconda3/envs/Refactor/lib/python3.9/site-packages/pytorch_lightning/utilities/cli.py", line 230, in parse_args
    argv = self._convert_argv_issue_84(classes, k, argv)
  File "/home/clear/fbartocc/miniconda3/envs/Refactor/lib/python3.9/site-packages/pytorch_lightning/utilities/cli.py", line 272, in _convert_argv_issue_84
    class_path = passed_args[f"{argv_key}.class_path"]
KeyError: '--optimizer.class_path'
```

### Environment

Output of your collect_env_details.py:

* Packages:
        - numpy:             1.21.2
        - pyTorch_debug:     False
        - pyTorch_version:   1.10.0
        - pytorch-lightning: 1.5.2
        - tqdm:              4.62.3
        - jsonargparse[signatures]:              3.19.3
* System:
        - OS:                Linux
        - architecture:
                - 64bit
                - ELF
        - processor:         x86_64
        - python:            3.9.7
        - version:           18.04.1-Ubuntu

- PyTorch installed with `conda` 
- I also tried with `jsonargparse 4.0.0` but the error is the same

Am I missing something ?

cc @carmocca @mauvilsa

**Comment by rohitgr7** [comment]
cc @carmocca

**Comment by tchaton** [comment]
@mauvilsa

**Comment by carmocca** [comment]
This is caused by the limitations of our "hack" implementation for https://github.com/omni-us/jsonargparse/issues/84. We might be able to extend the hack to support this, but IMO any effort by Mauricio would be better spent into actually implementing the feature natively.

**Comment by carmocca** [comment]
@F-Barto This will be fixed with #10767 which should be available with the Lightning 1.5.4 release next week. It will also require jsonargparse>=4.0.4 which has not been released yet.
